roles/nova-common/templates/etc/serverspec/nova_spec.rb

```diff
@@ -55,10 +55,10 @@
 
 describe file('/etc/nova/nova.conf') do
   it { should contain 'debug = {{ nova.logging.debug }}' }
-end 
+end
 
-describe file( '{{ nova.state_path }}/instances' ) do
+has_file = file('{{ nova.state_path }}/instances').exists?
+describe file( '{{ nova.state_path }}/instances' ), :if => has_file do
   it { should exist }
   it { should be a directory }
```

Comment:
That should exist inside doesn't need to be there. That directory is only created upon the creation of vm instance.